streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -23,11 +23,13 @@
 import java.util.Map;
 
 /**
- * The window specifications used for joins.
+ * The window specifications used for joins. TODO extend
  */
 public class JoinWindows extends Windows<TumblingWindow> {
 
+    /** Maximum time difference for tuples that are before the join tuple. */
```

Comment:
I guess it depends on the timestamp extractor you are using... `JoinWindows` are agnostic to the unit... Thus, for wall clock time is would be milliseconds. But if somebody assigned "seconds" as timestamps it would be seconds. @guozhangwang can you confirm?